pkgs/development/em-modules/generic/default.nix

```diff
@@ -0,0 +1,60 @@
+{ pkgs, lib, emscripten }:
+
+{ buildInputs ? [], nativeBuildInputs ? []
+
+, enableParallelBuilding ? true
+
+, meta ? {}, ... } @ args:
+
+pkgs.stdenv.mkDerivation (
+  args // 
+  {
+
+  name = "emscripten-${args.name}";
+  buildInputs = [ emscripten ] ++ buildInputs;
+  nativeBuildInputs = [ emscripten ] ++ nativeBuildInputs;
+
+  # fake conftest results with emscripten's python magic
+  EMCONFIGURE_JS=2;
+
+  configurePhase = args.configurePhase or ''
```

Comment:
you are referring to:

* configurePhase
* buildPhase
* checkPhase

not sure what you mean with that, but this is just a default implementation which should be overriden intentionally by emscriptionPackages.